### PR TITLE
sysbuild: Add support for sysbuild-assigned MCUboot image IDs

### PR DESCRIPTION
**Pull request by nordicjm**
Also includes changes from manifest: MCUboot KMU support for ED25519 signature verification https://github.com/nrfconnect/sdk-nrf/pull/17616 and treewide: update thingy91x bootloader setup https://github.com/nrfconnect/sdk-nrf/pull/18137 and nrf_compress: Fix ARM thumb filter cross-chunk issue https://github.com/nrfconnect/sdk-nrf/pull/18075

**Compliance/license failure expected due to edits to third party lzma library file**

test_sdk_mcuboot: sdk-nrf-pr-17567

### CODE REVIEW

File: modules/mcuboot/boot/zephyr/Kconfig
Comment (nvlsianpu) [inline modules/mcuboot/boot/zephyr/Kconfig:42]: `default 0xc000` become dead assignment

File: subsys/mcuboot_ids/Kconfig
Comment (nvlsianpu) [inline subsys/mcuboot_ids/Kconfig:1]: add help with something like (informative only, populated by the build-system) to each entry

File: modules/mcuboot/boot/zephyr/Kconfig
Comment (nordicjm) [reply modules/mcuboot/boot/zephyr/Kconfig:42]: it is the default for child/parent

File: subsys/mcuboot_ids/Kconfig
Comment (nordicjm) [reply subsys/mcuboot_ids/Kconfig:1]: why duplicate what is already there?

**Review by maxd-nordic — APPROVED** [review]

**Comment by gchwier** [comment]
About CI regression, tests from test-sdk-mcuboot passed from branch sdk-nrf-pr-17567.
I've added in PR description lin:
test-sdk-mcuboot: sdk-nrf-pr-17567
however jenkins is running from main branch.

**Comment by nordicjm** [comment]
@nrfconnect/ncs-co-build-system @nrfconnect/ncs-pluto @nrfconnect/ncs-code-owners @nrfconnect/ncs-aegir @nrfconnect/ncs-co-boards please review

File: subsys/mcuboot_ids/Kconfig
Comment (tomi-font) [inline subsys/mcuboot_ids/Kconfig:1]: Why the commented-out `depends on` in this file?

File: sysbuild/CMakeLists.txt
Comment (tomi-font) [inline sysbuild/CMakeLists.txt:326]: This line doesn't have a counterpart with `${DEFAULT_IMAGE}`. Intended?

File: sysbuild/Kconfig.mcuboot
Comment (tomi-font) [inline sysbuild/Kconfig.mcuboot:121]: Again why commented-out lines without a comment?

File: subsys/mcuboot_ids/Kconfig
Comment (tomi-font) [inline subsys/mcuboot_ids/Kconfig]: `if MCUBOOT` instead?

File: subsys/mcuboot_ids/Kconfig
Comment (nordicjm) [reply subsys/mcuboot_ids/Kconfig:1]: Those should have been removed, will be removed in next push

File: sysbuild/CMakeLists.txt
Comment (nordicjm) [reply sysbuild/CMakeLists.txt:326]: correct

File: sysbuild/Kconfig.mcuboot
Comment (nordicjm) [reply sysbuild/Kconfig.mcuboot:121]: Line will be removed in next push

**Review by michalek-no — APPROVED** [review]

File: subsys/mcuboot_ids/Kconfig
Comment (tejlmand) [inline subsys/mcuboot_ids/Kconfig:10]: I know sysbuild defined setting values will always take precedence over image defined values.

But is there no way we can avoid prompt settings purely for the purpose of sharing this info ?

side-note, `do not change` is strictly not required, cause if a user does change the value, then the value from sysbuild will overrule.

File: sysbuild/CMakeLists.txt
Comment (tejlmand) [inline sysbuild/CMakeLists.txt:177]: perhaps cherry-pick as `fromtree`: https://github.com/zephyrproject-rtos/zephyr/commit/8774f15bd2fe1afb70d22ecf669f399c5e1d1ffe
and adjust this line and others to: `set_config_int(mcuboot CONFIG_UPDATEABLE_IMAGE_NUMBER ${mcuboot_total_images})`

**Review by tejlmand — APPROVED** [review]
couple of observations, but nothing blocking

File: subsys/mcuboot_ids/Kconfig
Comment (nordicjm) [reply subsys/mcuboot_ids/Kconfig:10]: For sysbuild they will be forced, the warning however is for child/parent because they can freely be changed with it (which could break things). I think you submitted 2 different PRs to allow options to be set without a prompt from sysbuild -> images upstream but don't think either ever got finished

File: sysbuild/CMakeLists.txt
Comment (nordicjm) [reply sysbuild/CMakeLists.txt:177]: Will defer to after 2.8, do want to do a mass replace as `set_property()` is used in a lot of places and needs updating

File: subsys/mcuboot_ids/Kconfig
Comment (nordicjm) [reply subsys/mcuboot_ids/Kconfig]: Fixed

File: sysbuild/Kconfig.mcuboot
Comment (nordicjm) [reply sysbuild/Kconfig.mcuboot]: Fixed

File: subsys/mcuboot_ids/Kconfig
Comment (nordicjm) [reply subsys/mcuboot_ids/Kconfig:1]: Fixed

**Comment by nordicjm** [comment]
@nrfconnect/ncs-co-boards @nrfconnect/ncs-co-networking @nrfconnect/ncs-modem ncs-modem @nrfconnect/ncs-nrf-cloud ncs-nrf-cloud @nrfconnect/ncs-iot-oulu ncs-iot-oulu please review

**Comment by nordicjm** [comment]
@nrfconnect/ncs-iot-oulu please review

**Review by tomi-font — APPROVED** [review]

File: applications/serial_lte_modem/sysbuild/mcuboot/boards/thingy91x_nrf9151.overlay
Comment (MarkusLassila) [inline applications/serial_lte_modem/sysbuild/mcuboot/boards/thingy91x_nrf9151.overlay:3]: Is there a reason for not having this in https://github.com/nrfconnect/sdk-nrf/blob/main/applications/serial_lte_modem/boards/thingy91x_nrf9151_ns.overlay?

**Review by MarkusLassila — APPROVED** [review]

File: applications/serial_lte_modem/sysbuild/mcuboot/boards/thingy91x_nrf9151.overlay
Comment (nordicjm) [reply applications/serial_lte_modem/sysbuild/mcuboot/boards/thingy91x_nrf9151.overlay:3]: @maxd-nordic

**Review by eivindj-nordic — APPROVED** [review]

File: subsys/nrf_compress/lzma/armthumb.c
Comment (nordicjm) [reply subsys/nrf_compress/lzma/armthumb.c]: Fixed, the other issues with this file will be ignored as it is a third party file with a specific allowance

**Review by nvlsianpu — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**
